src/components/common/Clickable.d.ts

```diff
@@ -0,0 +1,15 @@
+import type { InteractiveProps } from 'src/components/Interactive'
+
+
+export type ClickableProps<T extends keyof JSX.Element = 'div'> = {
+  as?: T
+  href?: string
+  tooltip?: string
+  tooltipSide?: 'top' | 'bottom' | 'left' | 'right'
+  tooltipDelay?: number
+  useTooltipAsLabel?: boolean
+} & Omit<InteractiveProps<T>, 'as'>
+
+declare const Clickable: <T extends keyof JSX.Element>(props: ClickableProps<T>) => JSX.Element
+
+export default Clickable
```

Comment:
Hmm... how do the two Clickable exports get resolved at import time, and why is one in a js file not ts?

Reply:
This is a [type declaration file](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html). It allows adding types for a module without converting that module to TS. As long as the names match (module.d.ts goes with module.js), TypeScript will look here for the types.

This is generally how packages distributed via NPM provide types: they ship JS modules and type declarations as opposed to TS modules. And [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) provides type declarations for many libraries that don't provide their own. Terra UI gets types for several packages from DefinitelyTyped (the `@types/...` packages).

Reply:
Originally, this was a .d.ts file to avoid converting common.js to TS. Now that these have been split out, these could be converted to TS. But moving the existing JS functions and type declarations was the most expedient way to resolve the issues with TypeScript complaining about importing from common.js other components that were not in the type declarations file.